core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -7,19 +7,23 @@ class SemiCollapsibleNavigationClass {
     this.responsiveNavSetting = LuigiConfig.getConfigValue(
       'settings.responsiveNavigation'
     );
-    let currentCollapsedState = localStorage.getItem(
-      NavigationHelpers.COL_NAV_KEY
+    let currentCollapsedState = JSON.parse(
+      localStorage.getItem(NavigationHelpers.COL_NAV_KEY)
     );
-    let getResponsiveNavSettings =
+    const isResponsiveNavSemiCollapsibleOrFiori3 =
       this.responsiveNavSetting === 'semiCollapsible' ||
       this.responsiveNavSetting === 'Fiori3';
 
     //checking if here was a previous state in localStorage before the reload
-    if (currentCollapsedState != false && getResponsiveNavSettings) {
+    if (
+      currentCollapsedState != false &&
```

Comment:
Besides the naming, Please change to `!==`

Reply:
done